Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYApplePayPaymentProvider.h

```diff
@@ -0,0 +1,75 @@
+//
+//  BUYApplePayPaymentProvider.h
+//  Mobile Buy SDK
+//
+//  Created by Shopify.
+//  Copyright (c) 2015 Shopify Inc. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+@import UIKit;
+#import "BUYPaymentProvider.h"
+
+extern NSString *const _Nonnull BUYApplePayPaymentProviderId;
+
+@class BUYClient;
+
+@interface BUYApplePayPaymentProvider : NSObject <BUYPaymentProvider>
+
+/**
+ *  Initializer for Apple Pay payment provider
+ *
+ *  @param client     a `BUYClient`
+ *  @param merchantID the merchant ID for Apple Pay
+ *
+ *  @return an instance of `BUYApplePayPaymentProvider`
+ */
+- (instancetype _Nullable)initWithClient:(BUYClient * _Nonnull)client merchantID:(NSString * _Nonnull)merchantID NS_DESIGNATED_INITIALIZER;
+- (instancetype _Nullable)init NS_UNAVAILABLE;
+
+/**
+ *  The supported credit card payment networks.  Default values:
+ *  iOS 8.3: PKPaymentNetworkAmex, PKPaymentNetworkMasterCard, PKPaymentNetworkVisa.
+ *  iOS 9.0: PKPaymentNetworkAmex, PKPaymentNetworkDiscover, PKPaymentNetworkMasterCard, PKPaymentNetworkVisa.
+ */
+@property (nonatomic, copy) NSArray * _Nonnull supportedNetworks;
+
+/**
+ *  The merchant ID required for Apple Pay
+ */
+@property (nonatomic, copy, readonly) NSString * _Nonnull merchantID;
```

Comment:
Perhaps we could wrap the header in `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` ? It seems most values are non nil.